python/setup.py

```diff
@@ -231,7 +232,10 @@ def _run_cmake(self):
                 print("-- Runnning cmake for pyarrow")
                 self.spawn(cmake_command)
                 print("-- Finished cmake for pyarrow")
-                args = ['make']
+                if os.environ.get('PYARROW_CMAKE_GENERATOR') == 'Ninja':
```

Comment:
Actually you should be able to use `cmake --build .` as on Windows below.

Reply:
I guess then it's enough to keep the `else` branch (with minor modifications).